### PR TITLE
Converting to plugin

### PR DESCRIPTION
**Pull request by rutaihwa**
[ Assignment 2 - Converting to plugin ] #43

### CODE REVIEW

File: lib/version.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        config: {
+	    description: 'Returns the version of the server',
+	    handler: function (request, reply) {
+		
+                return reply({ version: Package.version });
+	    }
+        }
+    });
+    
+    next();
```
Comment (AdriVanHoudt) [inline lib/version.js]: `next` is a callback so better to do `return next()`

File: lib/index.js [hunk 2]
```diff
@@ -3,7 +3,7 @@
 var Hapi = require('hapi');
 var Hoek = require('hoek');
 var Package = require('../package.json');
```
Comment (AdriVanHoudt) [inline lib/index.js]: you are not using package in index anymore so no need to require it

Comment (rutaihwa) [inline lib/index.js]: Yeah, i overlooked that. Thanks @AdriVanHoudt

Comment (AdriVanHoudt) [inline lib/index.js]: np :+1:

File: lib/index.js [hunk 3]
```diff
@@ -2,8 +2,7 @@
 
 var Hapi = require('hapi');
 var Hoek = require('hoek');
-var Package = require('../package.json');
-
+var versionPlugin = require('./version');
```
Comment (hueniverse) [inline lib/index.js]: `Version`

File: lib/index.js [hunk 4]
```diff
@@ -15,16 +14,13 @@ internals.init = function () {
...
 
-    server.route({
-        method: 'GET',
-        path: '/version',
-        config: {
-            description: 'Returns the version of the server',
-            handler: function (request, reply) {
+    server.register({register: versionPlugin }, {
+	routes: {
+            prefix: '/version'
```
Comment (hueniverse) [inline lib/index.js]: Why use the prefix here instead of just setting the route path directly?

Comment (rutaihwa) [inline lib/index.js]: Incase, we'd have different versions like v1, v2 etc?

File: lib/index.js [hunk 5]
```diff
@@ -15,16 +14,9 @@ internals.init = function () {
...
-            handler: function (request, reply) {
+    server.register({register: Version }, {} , function (err) {
 
-                return reply({ version: Package.version });
-            }
-        }
+        Hoek.assert(!err, err);
     });
 
     server.start(function (err) {
```
Comment (edwin-bluekite) [inline lib/index.js]: Same as my error, you should start the server in the register callback

File: lib/version.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+	    description: 'Returns the version of the server',
+	    handler: function (request, reply) {
+		
+                return reply({ version: Package.version });
+	    }
+        }
+    });
+    
+    return next();
+}
```
Comment (edwin-bluekite) [inline lib/version.js]: Add semicolon

File: lib/version.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+
+exports.register.attributes = {
+    name: 'versionPlugin',
+    version: '0.0.1'
```
Comment (edwin-bluekite) [inline lib/version.js]: Version is not needed here

File: lib/index.js (hunk 4 above)
Comment (hueniverse) [inline lib/index.js]: This is a lot of complexity for no gain. Don't optimize for unknown future needs.

File: lib/version.js (hunk 6 above)
Comment (rutaihwa) [inline lib/version.js]: @edwin-bluekite where am I missing the semicolon again?

File: lib/index.js (hunk 4 above)
Comment (rutaihwa) [inline lib/index.js]: You are right! :+1:  Thanks.

**Pull request closed without merging**